src/components/PeopleFilters.tsx

```diff
@@ -1,21 +1,68 @@
+import { useNavigate, useSearchParams } from 'react-router-dom';
+import cn from 'classnames';
+import { SearchLink } from './SearchLink';
+import { getSearchWith } from '../utils/searchHelper';
+
 export const PeopleFilters = () => {
+  const navigate = useNavigate();
+  const [
+    searchParams,
+  ] = useSearchParams();
+
+  const onSetQuery = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const newQuery = {
+      query: event.target.value || null,
+    };
+
+    navigate({ search: getSearchWith(searchParams, newQuery) });
+  };
+
+  const currentCentury = searchParams.getAll('centuries');
+  const currentSex = searchParams.get('sex');
+  const currentQyery = searchParams.get('query');
+
   return (
     <nav className="panel">
       <p className="panel-heading">Filters</p>
 
       <p className="panel-tabs" data-cy="SexFilter">
-        <a className="is-active" href="#/people">All</a>
-        <a className="" href="#/people?sex=m">Male</a>
-        <a className="" href="#/people?sex=f">Female</a>
+        <SearchLink
+          params={{ sex: null }}
+          className={cn(
+            { 'is-active': currentSex === null },
+          )}
+        >
+          All
+        </SearchLink>
+
+        <SearchLink
+          params={{ sex: 'm' }}
+          className={cn(
+            { 'is-active': currentSex === 'm' },
+          )}
+        >
+          Male
+        </SearchLink>
+
+        <SearchLink
+          params={{ sex: 'f' }}
+          className={cn(
+            { 'is-active': currentSex === 'f' },
+          )}
+        >
+          Female
+        </SearchLink>
       </p>
 
       <div className="panel-block">
         <p className="control has-icons-left">
           <input
+            value={currentQyery || ''}
```

Comment:
```suggestion
            value={currentQuery || ''}
```

i thinks you mean `currentQuery`